### ISSUE TITLE
Fix false positives for hex colours in unit-whitelist

### ISSUE THREAD

**Issue by Bob-chemist** [stylelint/stylelint#4493]
> Clearly describe the bug

unit-whitelist doesn't properly parse hex colors, starting with number in border property.
If hex color contains any letter except for "E" in border property, it causes error. 
e.g. `border: 2px solid #2196F3`
If we move color to the first place of border property (border:  #2196F3 2px solid) or extract it to border-color property or change it to rgb(33, 150, 243), error dissappears. 
Single "E" in hex color dont't cause error because it is still a number - 2196E3 = 2196*10^3. "E" on last place or second "E" throws error too - it is not a proper number now.

> Which rule, if any, is the bug related to?

 `unit-whitelist`

> What CSS is needed to reproduce the bug?

```css
border: 2px solid #2196F3
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {   
    "unit-whitelist": [
      "px"
    ]
  }
}
```

> Which version of stylelint are you using?

`12.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

VS Code and stylelint extension

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"I use SASS syntax without braces"

> What did you expect to happen?

"No errors to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
filter.module.sass
  Unexpected unit "F3" (unit-whitelist) stylelint(unit-whitelist) [54, 25]

```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

As I see in demo it happens only to Sass syntax: [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIoEsBuAdAdgAmwIwgCdZD5sAmABwA9sBnCAG2SmwGJyBGATgDYAxAMxYQAGhB0AnpgAuAQ2oIJcunTHgImAGbIA5kuBZc6EIQCujGHRNlgubEfsmzmZDIC0AdwAWbmMzoZG2wAbUd7BxAaE3DsAF1HAF8sRJBEoA). Doesn't happen in standard CSS ([demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIoEsBuACYAdAdu9ARhAE6zHzoBMADgB7oDOEANslOgMSUCMAnAGwAxAMy4AviAA0IBgE8cAFwCGtBOAYMp4CDgBmyAOZrsedFhDEArsxgNzFYPjOn85yzmQKAtAHcAFp4wrAwK9ugA2rhOTuZ05lFOALoJYuIgYkA)) or SugarSS.

**Comment by fanich37** [comment]
It seems that the issues is caused by `postcss-sass` parser. When `walkDecls` is invoked it returns declaration with a value `2px solid 2196F3`.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider opening an issue upstream and fixing the issue there as the syntax is no longer bundled by Stylelint.
